### PR TITLE
Feature/24981 search criteria

### PR DESCRIPTION
**Pull request by n-singh**
Search criteria:
- departments
- persons
- phone numbers
- email address
- jobtitles

### CODE REVIEW

File: frontend/src/api/Search.js [hunk 1]
```diff
@@ -35,5 +35,17 @@ export default {
       .catch(error => {
         console.log(error.response)
       })
+  },
+
+  treeView (key, value) {
+    key = key || '*'
```
Comment (andersjepsen) [inline frontend/src/api/Search.js]: you don't use key and value for anything.

File: frontend/src/components/VSearchOption.vue [hunk 2]
```diff
@@ -32,6 +32,12 @@ export default {
     return {
       selected: null
     }
+  },
+
+  watch: {
+    selected (val) {
+      this.$emit('change-option', val)
```
Comment (andersjepsen) [inline frontend/src/components/VSearchOption.vue]: I think you can just emit on input, and then get the value with v-model where you use the v-search-option component? I.e. no need to create a new event

File: frontend/src/components/VTreeView/_store/index.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+const state = {
```
Comment (andersjepsen) [inline frontend/src/components/VTreeView/_store/index.js]: this entire store is a mistake by me I haven't removed yet. It's not being used. So just remove this store :)

File: frontend/src/store/modules/searchResults.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
+import Vue from 'vue'
```
Comment (andersjepsen) [inline frontend/src/store/modules/searchResults.js]: You're not using your state anywhere. Is it work in progress?

Comment (n-singh) [reply frontend/src/store/modules/searchResults.js]: Yes, it is

File: frontend/src/components/VSearchOption.vue (hunk 2 above)
Comment (n-singh) [reply frontend/src/components/VSearchOption.vue]: It, does'nt work because I use the event as a method

**Pull request merged**